### PR TITLE
[build] Improve development build process

### PR DESCRIPTION
**Pull request by aravindhp**
Introduce Dockerfile.base that builds only the submodules. This base image is used in Dockerfile.wmco which results in only the WMCO being built.

Now that we have simplified the image build process normalize on using `$OPERATOR_IMAGE` as the source of truth for the WMCO image.

### CODE REVIEW

**Review by jrvaldes — APPROVED** [review]
/lgtm

File: docs/HACKING.md
Comment (sebsoto) [inline docs/HACKING.md:22]: It would be nice if a make target was added that did this.
So then you could just do something like this:
`make dev-build IMG=quay.io/<insert username>/wmco:$VERSION_TAG`

Comment (sebsoto) [reply docs/HACKING.md:22]:
```
.PHONY: dev-build
dev-build:
        podman build . -t wmco-base:latest -f build/Dockerfile.base
        podman build . -t $(IMG) -f build/Dockerfile.wmco
        podman push $(IMG)
```

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: docs/HACKING.md
Comment (aravindhp) [reply docs/HACKING.md:22]: The main reason for the split Dockerfiles was so that you don't keep rebuilding the submodules which is exactly what this `Makefile` endpoint you are asking for is doing. IOW the endpoint is as good as doing `podman build . -t $(IMG) -f build/Dockerfile; podman push $(IMG)`. So I don't see how it helps us. So if you want `Makefile` endpoints it would be `dev-base-build`, `dev-wmco-build`, `dev-img-push`. Alternatively looking more at the new `Makefile`, I am wondering if `make run` with a local `/payload` directory is good enough. At which point, what we should do is add a `payload` endpoint.

Comment (aravindhp) [reply docs/HACKING.md:22]: >I am wondering if make run with a local /payload directory is good enough. At which point, what we should do is add a payload endpoint.

Just remembered that this won't work for us as WMCO needs to run on the private subnet.

**Comment by mansikulkarni96** [comment]
changes LGTM, please remove hold when you are ready.

File: docs/HACKING.md
Comment (sebsoto) [inline docs/HACKING.md:6]: `push the`

File: hack/olm.sh
Comment (sebsoto) [inline hack/olm.sh:55]: Thank you for changing this :smile:

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: docs/HACKING.md
Comment (aravindhp) [reply docs/HACKING.md]: Fixed

File: docs/HACKING.md
Comment (jrvaldes) [inline docs/HACKING.md:8]: Since `$VERSION_TAG` is a user defined field like `<insert username>`, I feel it should be written as `<version tag>`

**Review by jrvaldes — APPROVED** [review]
/lgtm

**Comment by mansikulkarni96** [comment]
LGTM
/hold for addressing @jrvaldes comment.

File: docs/HACKING.md
Comment (aravindhp) [reply docs/HACKING.md:8]: I am not introducing the `$VERSION_TAG` in this PR but is existing usage. Changes like this warrants a separate PR given we are using `$*_TAG` in multiple places and should be changed across the board.

**Pull request merged**
